### PR TITLE
Remove certificate hash from Artifact

### LINKED ISSUE #932: Remove certificate hash from Artifact
## Issue
Remove the certificate hash from the:
- Snapshot artifact entity
- Mithril Stake Distribution artifact entity
 
## To do

- [x] Remove `certificate_hash` in Snapshot entity
- [x] Remove `certificate_hash` in Mithril Stake Distribution entity
- [x] Create migration to remove `certificate_hash` from `signed_entity` table
- [x] Split `Message Adapter` in `From Message Adapter` (without type constraint) and `To Message Adapter` (with type constraint Serialize/DeSerialize)
- [x] Implement the correct `Message Adapter` trait for all the adapters in all the nodes

### PR DESCRIPTION
**Pull request by dlachaumepalo**
## Content
This PR includes updates of Snapshot and MithrilStakeDistribution entities.
- `certificate_hash` and `created_at` were removed in Snapshot entity
- `certificate_hash` and `created_at` were removed in MithrilStakeDistribution entity
- A migration was created to remove `certificate_hash` of `artifact` JSON in `signed_entity` table
- `MessageAdapter` trait was splitted in `FromMessageAdapter` and `ToMessageAdapter`
- Implement appropriate `MessageAdapter` trait for Mithril Client and Mithril Aggregator

## Pre-submit checklist

- Branch
  - [ ] Tests are provided (if possible)
  - [X] Crates versions are updated (if relevant)
  - [X] Commit sequence broadly makes sense
  - [X] Key commits have useful messages
- PR
  - [X] No clippy warnings in the CI
  - [X] Self-reviewed the diff
  - [X] Useful pull request description
  - [X] Reviewer requested
- Documentation
  - [ ] Update README file (if relevant)
  - [ ] Update documentation website (if relevant)
  - [ ] Add dev blog post (if relevant)

## Comments
`FromMithrilStakeDistributionListAdapter` and `FromSnapshotListMessageAdapter` adapters were removed since there are no longer used.

## Issue(s)
Closes #932

### CODE REVIEW

File: mithril-client/src/commands/snapshot/show.rs [hunk 1]
```diff
@@ -27,39 +30,51 @@ impl SnapshotShowCommand {
         let config: Config = config_builder.build()?;
         let mut dependencies_builder = DependenciesBuilder::new(Arc::new(config));
         let snapshot_service = dependencies_builder.get_snapshot_service().await?;
-        let snapshot = snapshot_service.show(&self.digest).await?;
+        let signed_entity = snapshot_service.show(&self.digest).await?;
 
         if self.json {
-            println!("{}", serde_json::to_string(&snapshot)?);
+            println!("{}", serde_json::to_string(&signed_entity.artifact)?);
```
Comment (Alenar) [inline mithril-client/src/commands/snapshot/show.rs:36]: Loss of the two removed fields in the json output, maybe use a SnapshotMessage instead ? (This means using a adapter)

File: mithril-common/src/entities/mithril_stake_distribution.rs [hunk 2]
```diff
@@ -36,31 +29,29 @@ impl MithrilStakeDistribution {
...
+    /// Do not add other parameters to the compute hash.
+    /// MSD is defined by the epoch and signers
     fn compute_hash(&self) -> String {
         let mut hasher = Sha256::new();
         hasher.update(self.epoch.0.to_be_bytes());
         for signer_with_stake in &self.signers_with_stake {
             hasher.update(signer_with_stake.compute_hash().as_bytes());
         }
-        hasher.update(self.certificate_hash.as_bytes());
+        // hasher.update(self.certificate_hash.as_bytes());
```
Comment (Alenar) [inline mithril-common/src/entities/mithril_stake_distribution.rs:54]: You can just remove this line :).

File: mithril-common/src/entities/mithril_stake_distribution.rs [hunk 3]
```diff
@@ -36,31 +29,29 @@ impl MithrilStakeDistribution {
...
-            certificate_hash,
-            created_at,
             protocol_parameters: protocol_parameters.to_owned(),
         };
         mithril_stake_distribution.hash = mithril_stake_distribution.compute_hash();
         mithril_stake_distribution
     }
 
+    /// Do not add other parameters to the compute hash.
+    /// MSD is defined by the epoch and signers
```
Comment (Alenar) [inline mithril-common/src/entities/mithril_stake_distribution.rs:47]: Can you replace the abbreviation with the full word ?

File: mithril-client/src/services/mithril_stake_distribution.rs [hunk 4]
```diff
@@ -153,24 +153,22 @@ impl MithrilStakeDistributionService for AppMithrilStakeDistributionService {
         genesis_verification_key: &str,
     ) -> StdResult<PathBuf> {
         // 1 - retrieve stake distribution
```
Comment (jpraynaud) [inline mithril-client/src/services/mithril_stake_distribution.rs:155]: Maybe you could also remove all these numbered comments that don't bring a lot here

File: mithril-aggregator/src/database/migration.rs [hunk 5]
```diff
@@ -548,5 +548,18 @@ drop table db_version;
 alter table new_db_version rename to db_version;
             ",
         ),
+        // Migration 16
+        // Update `signed_entity` to remove `certificate_hash` of `artifact` JSON.
```
Comment (jpraynaud) [inline mithril-aggregator/src/database/migration.rs:552]: Small fix here:
'// Update `signed_entity` table to remove `certificate_hash` and `created_at` from `artifact` JSON field.'

**Review by jpraynaud — APPROVED** [review]
LGTM :+1: 

I left few comments to fix before merging

**Pull request merged**